protocols/kad/src/behaviour.rs

```diff
@@ -283,6 +315,18 @@ impl<TSubstream> Kademlia<TSubstream> {
         self.start_query(QueryInfoInner::GetProviders { target, pending_results: Vec::new() });
     }
 
+    /// Starts an iterative `GET_VALUE` request.
+    pub fn get_value(&mut self, key: Multihash) {
+        self.start_query(QueryInfoInner::GetValue { key, results: vec![] });
```

Comment:
It would be good if `get_value` always first checks its local storage and if `Q=1` (which we can start with and may even have as the default), then we can short-circuit and immediately queue a `KademliaOut::GetValueRes`, without issuing a query, provided that we don't constrain ourselves to always return closest peers in the result, but only when the value was not found, as I suggest in another comment.

Reply:
The ability to short-circuit when first looking into `self.records` also provides an argument for taking `key` by reference here, because it may then actually not need to be cloned (only if the key is not found in `self.records` or `Q > 1`.